### ISSUE TITLE
Cover info should be displayed in a dedicated part

### ISSUE THREAD

**Issue by pjotrek-b** [MediaArea/MediaInfoLib#2288]
I've accidentally discovered that the following video (available to download from Vimeo) contains a strange additional video track that Mediainfo does *not* mention at all: :open_mouth: 
https://vimeo.com/255370370

But ffprobe does:
`Stream #0:3: Video: mjpeg, yuvj420p(pc, bt470bg/unknown/unknown), 1920x1080 [SAR 72:72 DAR 16:9], 90k tbr, 90k tbn, 90k tbc`

I've managed to extract this MJPEG track and it seems to be a single JPEG image.

However, I did not find any hint for this object in Mediainfo.

I've attached the [Text-output of MediaInfo](https://github.com/MediaArea/MediaInfo/files/2972416/winterschool-mediainfo.txt), as well as the [overview listing of ffprobe](https://github.com/MediaArea/MediaInfo/files/2972414/winterschool-ffprobe.txt).

When I extract only that MJPEG video track to Matroska, like this:
`$ ffmpeg -i winterschool_promo-2018.mov -map 0:3 -c copy delme.mkv`

...MediaInfo now sees it and shows its metadata as expected:

> Video
> ID                                       : 1
> Format                                   : V_MJPEG
> Codec ID                                 : V_MJPEG
> Width                                    : 1 920 pixels
> Height                                   : 1 080 pixels
> Display aspect ratio                     : 16:9
> Frame rate mode                          : Constant
> Frame rate                               : 1 000.000 FPS
> Color space                              : YUV
> Default                                  : Yes
> Forced                                   : No
> Color range                              : Full
> Matrix coefficients                      : BT.470 System B/G

**Comment by cjee21** [comment]
> This is something planned for years, but always postponed to tomorrow :).

I guess the difficult part would be getting the thumbnail/cover image data to `MediaInfo_Inform.cpp`.

**Comment by JeromeMartinez** [comment]
@cjee21 no, it is not like that, you'll see how it is done soon, there will be a PR soon about that.

**Comment by cjee21** [comment]
> [@cjee21](https://github.com/cjee21) no, it is not like that, you'll see how it is done soon, there will be a PR soon about that.

I'm referring to https://github.com/MediaArea/MediaInfoLib/issues/2288#issuecomment-2860462004 which is a reply to a suggestion about displaying the image itself in HTML view.

**Comment by JeromeMartinez** [comment]
@cjee21 actually there is already a first version at https://github.com/MediaArea/MediaInfoLib/pull/2289/files, you may look at "Merge(MI, Stream_Image, 0, StreamPos_Last);" line.
All the covers/thumbnails I know except the ones in EXIF (which will be added to your EXIF PR after I finish QA on the other PR)and TIFF (linked to the EXIF parsing so later) are expected to be supported.

**Comment by cjee21** [comment]
> [@cjee21](https://github.com/cjee21) actually there is already a first version at https://github.com/MediaArea/MediaInfoLib/pull/2289/files,

Aware of this which displays image info as a separate track. But my above comment is regarding the other comment which I understand as suggesting the display of the image itself in HTML view.

**Comment by JeromeMartinez** [comment]
> which is a reply to a suggestion about displaying the image itself in HTML view.

Ha, OK, so:

> I guess the difficult part would be getting the thumbnail/cover image data to MediaInfo_Inform.cpp.

Yes and no. It is mainly that the image data is optionally stored ("MediaInfoLib::Config.Flags1_Get(Flags_Cover_Data_base64)" in the PR), but there could be an option in the GUI for it and in that case the internal HTML output could add it + custom HTML outputs could peek %Cover_Data% (but it would work if there is only one image, I need to see how to let users manage multiple covers, more long term).

**Comment by cjee21** [comment]
@JeromeMartinez I just realized MediaInfoLib already has functionality to store base64 of cover image. That makes it easier. I have a test here for displaying cover image in HTML: https://github.com/cjee21/MediaInfoLib/tree/cover

**Comment by cjee21** [comment]
Side note: Just noticed MediaInfo generates non-compliant HTML when multiple files are opened. It just appends multiple HTML documents together. However, the resulting HTML file still works.

**Comment by JeromeMartinez** [comment]
> I have a test here for displaying cover image in HTML

PR please

>  Just noticed MediaInfo generates non-compliant HTML when multiple files are opened. It just appends multiple HTML documents together. 

there is something to do in MediaInfoList.cpp, if you are motivated ;-) (else I'll do it later this month, it should not be complicated).
